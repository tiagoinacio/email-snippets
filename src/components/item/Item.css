.item {
    border: 1px solid silver;
    padding: 10px;
    margin-bottom: 10px;
    width: 100%;
    cursor: pointer;
    text-align: left;
    color: #2f4f4f;
}