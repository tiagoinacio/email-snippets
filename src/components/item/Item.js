import React from 'react';
import PropTypes from 'prop-types';
import './Item.css';

const Item = ({ template, id, ...props }) => (
    <button
        { ...props }
        title={ id }
        className="item">
        { template }
    </button>
);

Item.propTypes = {
    template: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
};

export default Item;