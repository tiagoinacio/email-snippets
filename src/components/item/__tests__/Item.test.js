import React from 'react';
import Item from '..';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('should render an <Item />', () => {
  const component = mount(
      <Item id="first" template="my template" />
  );

  expect(component).toMatchSnapshot();
});
