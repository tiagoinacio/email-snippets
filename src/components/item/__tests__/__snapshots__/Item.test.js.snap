// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an <Item /> 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Item
    id="first"
    template="my template"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "id": "first",
      "template": "my template",
    },
    "ref": null,
    "rendered": Object {
      "instance": <button
        title="first"
        class="item"
>
        my template
</button>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "my template",
        "className": "item",
        "title": "first",
      },
      "ref": null,
      "rendered": Array [
        "my template",
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "id": "first",
        "template": "my template",
      },
      "ref": null,
      "rendered": Object {
        "instance": <button
          title="first"
          class="item"
>
          my template
</button>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "my template",
          "className": "item",
          "title": "first",
        },
        "ref": null,
        "rendered": Array [
          "my template",
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
